Add docs for using Mariner-based images (#2549)
dapr/docs: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-deploy.md

dapr-sentry-9435776c7f-8f7yd             1/1       Running   0          40s
```

## Using Mariner-based images

> When deploying Dapr, on Kubernetes but also in Docker self-hosted, the default container images that are pulled are based on [*distroless*](https://github.com/GoogleContainerTools/distroless).

Reviewer: ```suggestion
When deploying Dapr, whether on Kubernetes or in Docker self-hosted, the default container images that are pulled are based on [*distroless*](https://github.com/GoogleContainerTools/distroless).
```

> As an alternative option, you can use Dapr container images based on Mariner 2 (minimal distroless). [Mariner](https://github.com/microsoft/CBL-Mariner/), officially known as CBL-Mariner, is a free and open source Linux distribution and container base image that is maintained by Microsoft. For some Dapr users, leveraging container images based on Mariner can help meeting compliance requirements.

Reviewer: ```suggestion
Alternatively, you can use Dapr container images based on Mariner 2 (minimal distroless). [Mariner](https://github.com/microsoft/CBL-Mariner/), officially known as CBL-Mariner, is a free and open-source Linux distribution and container base image maintained by Microsoft. For some Dapr users, leveraging container images based on Mariner can help them meet compliance requirements.
```

Author: How about "help you meet"?

Reviewer: That works for me too!

> To use Mariner-based images for Dapr, you need to add `-mariner` to your Docker tags. For example, while `ghcr.io/dapr/dapr:latest` is the Docker image based on *distroless*, `ghcr.io/dapr/dapr:latest-mariner` is based on Mariner. Tags pinned to a specific version are available too, such as `{{% dapr-latest-version short="true" %}}-mariner`.

Reviewer: ```suggestion
To use Mariner-based images for Dapr, you need to add `-mariner` to your Docker tags. For example, while `ghcr.io/dapr/dapr:latest` is the Docker image based on *distroless*, `ghcr.io/dapr/dapr:latest-mariner` is based on Mariner. Tags pinned to a specific version are also available, such as `{{% dapr-latest-version short="true" %}}-mariner`.
```
